CodenameOne/src/com/codename1/io/JSONParser.java

```diff
@@ -632,4 +634,270 @@ public void keyValue(String key, String value) {
+     * address2.put("state", "NY");
+     * address2.put("postalCode", "10021-3100");
+     * Map<String, Object> secondPerson = new LinkedHashMap<>();
+     * secondPerson.put("firstName", "Gioia");
+     * secondPerson.put("lastName", "Mia");
+     * secondPerson.put("isAlive", true);
+     * secondPerson.put("age", 34);
+     * secondPerson.put("weight (kg)", 60.2);
+     * secondPerson.put("address", address2);
+     * address2.put("phoneNumbers", phoneNumbers2);
+     * Map<String, Object> firstPerson = new LinkedHashMap<>();
+     * firstPerson.put("firstName", "Paco");
+     * firstPerson.put("lastName", "Bellèz");
+     * firstPerson.put("isAlive", true);
+     * firstPerson.put("age", 35);
+     * firstPerson.put("weight (kg)", 70.7);
+     * firstPerson.put("address", address1);
+     * firstPerson.put("partner", secondPerson);
+     * firstPerson.put("children", new ArrayList());
+     * firstPerson.put("extraInfo", null);
+     * firstPerson.put("phoneNumbers", phoneNumbers);
+     * List friends = new ArrayList();
+     * friends.add("Paul");
+     * friends.add("Karl");
+     * friends.add("Mary");
+     * firstPerson.put("onVacation", false);
+     * firstPerson.put("friends", friends);
+     *
+     * Log.p("--- mapToJson() test");
+     * try {
+     *     Log.p("\n" + mapToJson(firstPerson));
+     * } catch (Exception ex) {
+     *     Log.e(ex);
+     * }
+     * }</pre>
+     *
+     * The output will be:
+     * <pre>{@code
+     * {
+     *   "firstName" : "Paco",
+     *   "lastName" : "Bellèz",
+     *   "isAlive" : true,
+     *   "age" : 35,
+     *   "weight (kg)" : 70.7,
+     *   "address" : {
+     *     "streetAddress" : "53, London Street",
+     *     "city" : "Paris",
+     *     "state" : "FR",
+     *     "postalCode" : "54856"
+     *   },
+     *   "partner" : {
+     *     "firstName" : "Gioia",
+     *     "lastName" : "Mia",
+     *     "isAlive" : true,
+     *     "age" : 34,
+     *     "weight (kg)" : 60.2,
+     *     "address" : {
+     *       "streetAddress" : "21 2nd Street",
+     *       "city" : "New York",
+     *       "state" : "NY",
+     *       "postalCode" : "10021-3100",
+     *       "phoneNumbers" : [{
+     *         "mobile" : "06124578965"
+     *       }]
+     *     }
+     *   },
+     *   "children" : [],
+     *   "extraInfo" : null,
+     *   "phoneNumbers" : [{
+     *     "home" : "212 555-1234"
+     *   },{
+     *     "office" : "646 555-4567"
+     *   },{
+     *     "mobile" : "123 456-7890"
+     *   }],
+     *   "onVacation" : false,
+     *   "friends" : ["Paul","Karl","Mary"]
+     * }
+     * }</pre>
+     *
+     * @param map The map to be converted to a JSON string
+     * @return The JSON string
+     * @throws Exception A generic exception thrown when converting the map to a
+     * JSON string fails.
+     * @author Francesco Galgani
+     */
+    public static String mapToJson(Map<String, Object> map) throws Exception {
```

Comment:
- The argument should be `Map<String, ?>`.
- Throwing a checked exception is a big pain for such utils.